logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -146,7 +146,12 @@ private void innerWriteEntry(DLQEntry entry) throws IOException {
 
     @Override
     public synchronized void close() throws IOException {
-        this.lock.release();
+        if (this.lock != null){
+            this.lock.release();
+            if (this.lock.channel() != null && this.lock.channel().isOpen()) {
```

Comment:
I agree around closeQuietly, but we don't have that on the classpath right now.
I will wrap the lock channel close inside a try... catch to alleviate your fears, but the channel has to be closed otherwise the delete cannot complete on Windows until the JVM exits.